src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -511,18 +511,20 @@ namespace winrt::Microsoft::Terminal::Control::implementation
     // - The automation peer for our control
     Windows::UI::Xaml::Automation::Peers::AutomationPeer TermControl::OnCreateAutomationPeer()
     {
-        if (_initializedTerminal && !_IsClosing()) // only set up the automation peer if we're ready to go live
+        // MSFT 33353327: We're purposefully not using _initializedTerminal to ensure we're fully initialized.
+        // Doing so makes us return nullptr when XAML requests an automation peer.
+        // Instead, we need to give XAML an automation peer, then fix it later.
+        if (!_IsClosing())
         {
             // create a custom automation peer with this code pattern:
             // (https://docs.microsoft.com/en-us/windows/uwp/design/accessibility/custom-automation-peers)
             if (const auto& interactivityAutoPeer{ _interactivity.OnCreateAutomationPeer() })
             {
-                auto margins{ SwapChainPanel().Margin() };
-
-                Core::Padding padding{ margins.Left,
-                                       margins.Top,
-                                       margins.Right,
-                                       margins.Bottom };
+                const auto margins{ SwapChainPanel().Margin() };

```

Comment:
Does the `SwapChainPanel()` already exist at this point?

Reply:
I think so. Here's my reasoning:
1. XAML creates the TermControl
   - `InitializeComponent()` is called, so we load TermControl.xaml
   - we `TermControl::_InitializeTerminal()` when `SwapChainPanel().LayoutUpdated([]{...})` later
2. XAML then calls `TermControl::OnCreateAutomationPeer()`
   - `SwapChainPanel()` exists, it's layout might not be up to date, however
3. _NOW_ `SwapChainPanel()`'s layout is updated, so we initialize the terminal
4. let's ensure the margin is up to date https://github.com/microsoft/terminal/blob/4d8771acc3402e31e53cbc622ceb208f50446ec2/src/cascadia/TerminalControl/TermControl.cpp#L722-L732